core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -106,6 +107,20 @@ final case class NonEmptyList[A](head: A, tail: List[A]) {
     toList.iterator.map(A.show).mkString("NonEmptyList(", ", ", ")")
 
   override def toString: String = s"NonEmpty$toList"
+
+  /**
+   * Remove duplicates. Duplicates are checked using `Order[_]` instance.
+   */
+  def distinct(implicit O: Order[A]): NonEmptyList[A] = {
```

Comment:
Is `Priority` recursive, i.e. can support > 2 implicits? Regarding hash sets and `Hash` typeclass, since hash sets implementations in Scala are based on `Object.hashcode()` it would require some wrapping of the elements, wouldn't it?